sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -147,6 +149,15 @@ case class BroadcastExchangeExec(
     } catch {
       case ex: TimeoutException =>
         logError(s"Could not execute broadcast in ${timeout.toSeconds} secs.", ex)
+        val executionUIData = sqlContext.sparkSession.sharedState.statusStore.
```

Comment:
Is this a reliable way to get the associated jobs?

Reply:
SQLAppStatusListener will hold the live execution data, so i use the same for getting the associated   jobs , If its not efficient way of getting then i will revisit the code and  try to find a better mechanism for getting the associated job for the particular execution Id. please let me know for any suggestions. thanks for your  valuable time.

Reply:
I cannot find an alternative to get Jobs based on the execution id in this layer of execution, seems to be this way is reliable as whenever we are submitting /processing events via dagscheduler, we are always posting the events to SQLAppStatusListener, this will make our job viable to retrieve the jobs from LiveExecutionData.
Please let me know  if we have any better way to get this job done. Thanks .